sdk/identity/Azure.Identity/src/ImdsManagedIdentityProbeSource.cs

```diff
@@ -0,0 +1,196 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Microsoft.Identity.Client.AppConfig;
+
+namespace Azure.Identity
+{
+    internal class ImdsManagedIdentityProbeSource : ManagedIdentitySource
```

Comment:
A lot of this class, especially the constant strings below are identical to `ImdsManagedIdentitySource`.
Is it possible to avoid some of this duplication?

It seems the key difference is removing `private bool _isFirstRequest = true;` and adding ` private MsalManagedIdentityClient _client;`

Reply:
`ImdsManagedIdentitySource` will go away entirely once we GA.